File: Number_Guess.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
    const int MIN = 1;
    const int MAX = 40;
    int guess = 0;
    int guesses = 7;
    int answer = 0;

    srand(time(0));

    answer = (rand() % MAX) + MIN;
    printf("*********\n");
    printf("Welcome to the number guessing game\n");
    printf("You need to guess the number between 1 and 40 before you run out of 7 lives\n");
    printf("Good luck...\n");
    printf("*********\n");

    do {
        printf("Enter a guess: ");
        scanf("%d", &guess);
        
        if (guess > answer) {
            printf("Please enter a lower value...\n");
        } else if (guess < answer) {
            printf("Please enter a higher value...\n");
        } else {
            printf("Congratulations\n");
            break;
        }
        
        guesses--;
        printf("Lives left: %d\n", guesses);

        if (guesses == 0) {
            printf("You have run out of guesses...\n");
            break;
        }
    } while (guess != answer);

    printf("**********************\n");
    printf("The correct number was: %d\n", answer);
    printf("**********************\n");

    return 0;
}

File: Number_Guess.py
import random
import time

print("""*****************************

    Number Guessing Game

    Guess a number between 1 and 40.

    before you run out of 7 lives


    *****************************""")

random_number = random.randint(1, 40)
guess_limit = 7

while guess_limit > 0:

    try:
        guess = int(input("Guess: "))
    except:
        print("Please enter a valid integer.")
        continue

    print("Checking...")
    time.sleep(1)

    if (guess < random_number):
        print("Please enter a higher value...")
    elif (guess > random_number):
        print("Please enter a lower value...")
    else:
        print("Congratulations!\nThe number is: ", random_number)
        break
    
    guess_limit -= 1
    print("Lives left:",guess_limit)

    if (guess_limit == 0):
        print("You have run out of guesses...")
        print("The number was: ", random_number)
